File: dataforge-fx/src/main/java/hep/dataforge/fx/values/ColorValueChooser.java
package hep.dataforge.fx.values;

import hep.dataforge.values.Value;
import javafx.scene.control.ColorPicker;
import javafx.scene.paint.Color;
import org.slf4j.LoggerFactory;

/**
 * Created by darksnake on 01-May-17.
 */
public class ColorValueChooser extends ValueChooserBase<ColorPicker> {
    @Override
    public void setDisplayValue(Value value) {
        if (!value.stringValue().isEmpty()) {
            try {
                getNode().setValue(Color.valueOf(value.stringValue()));
            } catch (Exception ex) {
                LoggerFactory.getLogger(getClass()).warn("Invalid color field value: " + value.stringValue());
            }
        }
    }

    @Override
    protected ColorPicker buildNode() {
        ColorPicker node = new ColorPicker();
        node.getStyleClass().add("split-button");
        node.setMaxWidth(Double.MAX_VALUE);
        node.setOnAction(event -> valueProperty().setValue(Value.of(node.getValue().toString())));
        return node;
    }
}

File: dataforge-gui/src/main/kotlin/hep/dataforge/fx/values/ColorValueChooser.kt
package hep.dataforge.fx.values

import hep.dataforge.values.Value
import javafx.scene.control.ColorPicker
import javafx.scene.paint.Color
import org.slf4j.LoggerFactory

/**
 * Created by darksnake on 01-May-17.
 */
class ColorValueChooser : ValueChooserBase<ColorPicker>() {
    override fun setDisplayValue(value: Value) {
        if (!value.stringValue().isEmpty()) {
            try {
                node.value = Color.valueOf(value.stringValue())
            } catch (ex: Exception) {
                LoggerFactory.getLogger(javaClass).warn("Invalid color field value: " + value.stringValue())
            }

        }
    }

    override fun buildNode(): ColorPicker {
        val node = ColorPicker()
        node.styleClass.add("split-button")
        node.maxWidth = java.lang.Double.MAX_VALUE
        node.setOnAction { _ -> value = Value.of(node.value.toString()) }
        return node
    }
}
